src/lib/viewers/media/__tests__/DashControls-test.tsx

```diff
@@ -48,67 +19,189 @@ describe('DashControls', () => {
-                onVolumeChange,
-            });
-
-            expect(wrapper.hasClass('bp-DashControls')).toBe(true);
-            expect(wrapper.find(MediaFullscreenToggle).prop('onFullscreenToggle')).toEqual(onFullscreenToggle);
-            expect(wrapper.find(MediaSettings).props()).toMatchObject({
-                audioTrack: 1,
-                audioTracks: [],
-                autoplay: false,
-                onAudioTrackChange,
-                onAutoplayChange,
-                onQualityChange,
-                onRateChange,
-                quality: 'auto',
-                rate: '1.0',
-            });
-            expect(wrapper.find(PlayPauseToggle).prop('onPlayPause')).toEqual(onPlayPause);
-            expect(wrapper.find(TimeControls).prop('onTimeChange')).toEqual(onTimeChange);
-            expect(wrapper.find(VolumeControls).prop('onMuteChange')).toEqual(onMuteChange);
-            expect(wrapper.find(VolumeControls).prop('onVolumeChange')).toEqual(onVolumeChange);
+            render(
+                <DashControls
+                    audioTrack={1}
+                    audioTracks={[]}
+                    autoplay={false}
+                    isPlaying={false}
+                    isPlayingHD={false}
+                    onAudioTrackChange={onAudioTrackChange}
+                    onAutoplayChange={onAutoplayChange}
+                    onFullscreenToggle={onFullscreenToggle}
+                    onMuteChange={onMuteChange}
+                    onPlayPause={onPlayPause}
+                    onQualityChange={onQualityChange}
+                    onRateChange={onRateChange}
+                    onTimeChange={onTimeChange}
+                    onVolumeChange={onVolumeChange}
+                    quality={Quality.AUTO}
+                    rate="1.0"
+                />,
+            );
+
+            await user.click(screen.getByTitle('Enter fullscreen'));
+            expect(onFullscreenToggle).toHaveBeenCalledTimes(1);
+
+            await user.click(screen.getByTitle('Play'));
+            expect(onPlayPause).toHaveBeenCalledTimes(1);
+
+            fireEvent.mouseDown(screen.getByLabelText('Media Slider'));
+            expect(onTimeChange).toHaveBeenCalledTimes(1);
+
+            await user.click(screen.getByTitle('Mute'));
+            expect(onMuteChange).toHaveBeenCalledTimes(1);
+
+            fireEvent.mouseDown(screen.getByLabelText('Volume Slider'));
+            expect(onVolumeChange).toHaveBeenCalledTimes(1);
         });
 
-        test.each([true, false])('should set isHDSupported prop on MediaSettings as %s', isHDSupported => {
-            const wrapper = getWrapper({ isHDSupported });
-            expect(wrapper.find(MediaSettings).prop('isHDSupported')).toBe(isHDSupported);
+        test.each([true, false])('should set isHDSupported prop on MediaSettings as %s', async isHDSupported => {
+            const user = userEvent.setup();
+            render(
+                <DashControls
```

Comment:
Should we extract it in order to not duplicate setting props in every test case?